### ISSUE TITLE
Brew Upgrade Shouldn't Berate Homebrew Users

### ISSUE THREAD

**Issue by gruberchris** [Homebrew/brew#16050]
### Verification

- [X] This issue's title and/or description do not reference a single formula e.g. `brew install wget`. If they do, open an issue at https://github.com/Homebrew/homebrew-core/issues/new/choose instead.

### Provide a detailed description of the proposed feature

Instead of berating Homebrew users, perhaps publish a blog article and be somewhat more constructive in your communication.

<img width="704" alt="image" src="https://github.com/Homebrew/brew/assets/2971844/c8dc56b7-1784-4948-916f-695fe6de6da2">

### What is the motivation for the feature?

To not be berated by Homebrew for using a slightly older MacBook from 2014.

### How will the feature be relevant to at least 90% of Homebrew users?

Most Homebrew users do not want to feel berated, unless they're into saddism.

### What alternatives to the feature have been considered?

Ditch berating Homebrew users and instead offer up a blog article discussing how best to share feature requests or to report bugs.

**Comment by MikeMcQuaid** [comment]
@gruberchris No-one is berating you here. We are fairly loudly (I'll admit that) pointing out that your macOS version is no longer support by Apple or Homebrew and that we don't want you to file issues (including this one!) complaining about that because it's a known design choice for us to not be able to support things that Apple can't.

I know it feels like overkill to you but: you're not running a project with millions of users with tens of volunteers scattered throughout the world. Everything mentioned there is something that we have had many problems with over a long period.

TL;DR: if we had the resources, time, money and support from Apple to be able to support every version of macOS forever: we would but we don't so we don't. Sorry!
